### PR TITLE
React Effects and Fetch: ISS Tracker

### PR DESCRIPTION
**Pull request by aclaes**
### 🔎 Review

Review via [🔗 **CodeSandbox**](https://codesandbox.io/s/github/neuefische/web-exercises/tree/react-effects-and-fetch_iss-tracker/sessions/react-effects-and-fetch/iss-tracker?file=/README.md) or locally by running this command in your Terminal:

```
npx ghcd@latest neuefische/web-exercises/tree/react-effects-and-fetch_iss-tracker/sessions/react-effects-and-fetch/iss-tracker -i
```

### ✨ Copyable Markdown for Curriculum

````md
Solve this challenge on
[🔗 **CodeSandbox**](https://codesandbox.io/s/github/neuefische/web-exercises/tree/main/sessions/react-effects-and-fetch/iss-tracker?file=/README.md)
or locally by running this command in your Terminal:

```
npx ghcd@latest neuefische/web-exercises/tree/main/sessions/react-effects-and-fetch/iss-tracker -i
```
````

### CODE REVIEW

File: sessions/react-effects-and-fetch/iss-tracker/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+We already have a state called `coords` in the `App` component which will hold the longitude and latitude values of the ISS position.
+
+1. Write the function `getISSCoords`. Fetch from the given url saved in the constant `URL`. After fetching successfully update the `coords` state accordingly.
+
+2. The refresh button already executes `getISSCoords` when pressed. Test if the function works as expected. (HINT: You can check if the position is displayed correctly by double checking on [this website](http://open-notify.org/Open-Notify-API/)).
+
+3. With the help of `useEffect` set an interval (with `setInterval`) to update the ISS coords every 5 seconds. The interval should be installed only on the first render of the app.
+   (HINT: Research `setInterval` on MDN if you don't know it)
+
+4. The `setInterval` function returns an ID of the respective interval. You can clear the interval by calling `clearInterval` with this id. Inside the useEffect, return a cleanup function, which executes this `clearInterval` function with the correct ID.
```
Comment (PReichetanz) [inline sessions/react-effects-and-fetch/iss-tracker/README.md:16]: I don't understand why we are doing this here: I see that the cleanup function is explained in the handout, but why is it useful in this specific example?

As far as I can see, the cleanup function removes the timer, so the data are not fetched every 5 seconds anymore. Why do we break the functionality we've just implemented? Only to train how to write a cleanup function?

If so, the question will raise what the purpose of this cleanup is. I think we should add an explanation here.

**Review by PReichetanz — COMMENTED** [review]
Nice and exciting exercise! Please find [my solution here](https://codesandbox.io/s/react-effects-and-fetch-iss-tracker-solution-01-p3g14b?file=/src/App.js).

I just don't understand the usage of the cleanup function and added a remark in the readme file.

File: sessions/react-effects-and-fetch/iss-tracker/README.md (hunk 1 above)
Comment (aclaes) [reply sessions/react-effects-and-fetch/iss-tracker/README.md:16]: I wrote a short explanation in the exercise description and additional not in the handout about the cleanup function

**Review by PReichetanz — APPROVED** [review]
That's exactly the note we need here, thanks!

**Pull request merged**